### PR TITLE
Add lines_t.s_max_pu to kmeans clustering

### PR DESCRIPTION
**Pull request by ClaraBuettner**
## Changes proposed in this Pull Request
We would like to use the time-varying s_max_pu attribute of lines to model weather-dependent line rating in [eTraGo](https://github.com/openego/eTraGo). The s_max_pu time series are calculated for each transmission line in a high spatial resolution. We are using PyPSA's kmeans_clustering to reduce the complexity, but this currently not includes time-varying line attributes. 

The clustering of time-varying line attributes is added to the kmeans_clustering in this branch. The clustered time series are weighted with the nominal power of each line. The implementation is very similar to the clustering of generator time series. 

## Checklist

- [ ] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.
- [ ] Unit tests for new features were added (if applicable).
- [ ] Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py` (if applicable).
- [ ] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.

### CODE REVIEW

**Comment by fneum** [comment]
Great, we've also discussed DLR recently. Do you mind me pushing directly to the branch to sync with master?

**Comment by ClaraBuettner** [comment]
> Great, we've also discussed DLR recently. Do you mind me pushing directly to the branch to sync with master?

Of course, you can push changes directly into the branch.

**Comment by fneum** [comment]
Seems like I don't have access to push to this branch. Would you mind merging the current master? This should run without conflicts and should trigger the tests.

**Comment by FabianHofmann** [comment]
After a short look: If you have a solved network, the clustering would lead to a weighted-average of the transmitted power, we should probably add a clustering strategy here.

File: pypsa/networkclustering.py [hunk 1]
```diff
@@ -207,9 +207,24 @@ def aggregatelinegroup(l):
...
+            df.columns = df.columns.astype(str)
+            lines_agg_b = df.columns.to_series().map(linemap).dropna()
+            df_agg = df.loc[:, lines_agg_b.index]
+            if not df_agg.empty:
+                pnl_df = df_agg.mul(
+                    network.lines.groupby(linemap).s_nom.apply(
+                        lambda grp: grp/grp.sum()
+                        )).groupby(linemap, axis=1).sum()
+                pnl_df.columns = _flatten_multiindex(pnl_df.columns).rename("name")
+                lines_t[attr] = pnl_df
```
Comment (FabianHofmann) [inline pypsa/networkclustering.py:223]:
```suggestion
    if with_time:
        for attr, df in network.lines_t.items():
            lines_agg_b = df.columns.to_series().map(linemap).dropna()
            df_agg = df.loc[:, lines_agg_b.index]
            if not df_agg.empty:
                if (attr == 's_max_pu') or (attr == "s_min_pu"):
                    weighting = network.lines.groupby(linemap).s_nom.transform(_normed)
                    df_agg = df_agg.multiply(weighting.loc[df_agg.columns], axis=1)
                pnl_df = df_agg.groupby(linemap, axis=1).sum()
                pnl_df.columns = _flatten_multiindex(pnl_df.columns).rename("name")
                lines_t[attr] = pnl_df

```

**Review by FabianHofmann — CHANGES_REQUESTED** [review]
Hey @ClaraBuettner, sorry that this took so long. Considering the earlier comment I suggest the following change

File: pypsa/networkclustering.py (hunk 1 above)
Comment (FabianHofmann) [reply pypsa/networkclustering.py:223]: @ClaraBuettner makes sense. indeed, I merged your contribution via a further PR. Thanks for tackling this.

**Pull request closed without merging**